Sync promotions added to Bookmarks & Passwords screens

amddg44 opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.

⚠️ If you're an external contributor, please file an issue first before working on a PR, as we can't guarantee that we will accept your changes if they haven't been discussed ahead of time. Thanks!
-->

Task/Issue URL: https://app.asana.com/0/72649045549333/1208017419957473/f
Tech Design URL:
CC:

**Description**:
New promo banners added + Sync screen updates to include link to download app on other devices

<!--
If at any point it isn't actively being worked on/ready for review/otherwise moving forward strongly consider closing it (or not opening it in the first place). If you decide not to close it, use Draft PR while work is still in progress or use `DO NOT MERGE` label to clarify the PRs state and comment with more information.
-->

**Steps to test this PR**:
### Sync Promo for Passwords screen
1. Starting with a clean install
2. Visit the Passwords screen and confirm the Sync Promo is not presented (since in empty state)
3. Create a saved credential manually 
4. Go back to the Passwords list screen and confirm the Sync promo is now presented
5. Confirm the promo is hidden when editing and searching
6. Tap "Set Up Sync" in the promo and confirm you are taken to the Sync screen, but do not enable Sync
7. Go back to Passwords & confirm the promo is still presented
8. Tap "No Thanks" in the promo card & confirm the promo is dismissed
9. Back out of the Passwords screen & return, confirming the promo does not get presented again
10. From the main Debug menu screen, scroll down to the very end and tap "Reset Sync Promo Prompts"
11. Confirm the promo is visible again in the Passwords screen

### Sync Promo for Bookmarks screen
1. Visit the Bookmarks screen and confirm the Sync Promo is not presented (since in empty state)
2. Save a bookmark 
3. Come back to the Bookmarks screen and confirm the Sync promo is now presented
4. Confirm the promo is hidden when editing and searching
5. Create a bookmark folder and tap into it, confirming the promo is not presented in the folder  
6. Back on the main Bookmarks screen, tap "Set Up Sync" in the promo 
7. Confirm you are taken to the Sync screen, but do not enable Sync
8. Go back to Bookmarks & confirm the promo is still presented
9. Tap "No Thanks" in the promo card & confirm the promo is dismissed
10. Back out of the Bookmarks screen & return, confirming the promo does not get presented again
11. From the main Debug menu screen, scroll down to the very end and tap "Reset Sync Promo Prompts"
12. Confirm the promo is visible again in the Bookmarks screen

### Sync screen + new "Get DuckDuckGo" screen
1. Go to the Sync screen
2. Confirm there is now a row "Get DuckDuckGo on Other Devices" in the Sync inactive state
3. Tap the row confirming the new "Get DuckDuckGo" screen is presented
4. Tap the blue link text "duckduckgo.com/app" and confirm copy to clipboard works
5. Tap the Share button and confirm the share sheet is presented
6. Go back to the Sync screen & start the flow to enable Sync (can be as a backup)
7. On the "Your data is synced!" screen confirm there is now a "Get DuckDuckGo on Other Devices" button below the "Done" button
8. Tap and confirm you are taken to the "Get DuckDuckGo" screen
9. Go back to the Sync screen & confirm the row "Get DuckDuckGo on Other Devices" is present under the Synced Devices section
10. Tap and confirm you are taken to the "Get DuckDuckGo" screen
11. Go back to the Bookmarks & Password screens and confirm the promo is not presented, now that Sync has been enabled


<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [x] Portrait
* [x] Landscape

**Device Testing**:

* [x] iPhone SE (1st Gen)
* [ ] iPhone 8
* [x] iPhone X
* [ ] iPhone 14 Pro
* [x] iPad

**OS Testing**:

* [x] iOS 15
* [ ] iOS 16
* [x] iOS 17

**Theme Testing**:

* [x] Light theme
* [x] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

amddg44 committed:
SyncPromotion feature flag added

amddg44 committed:
SyncPromoView created & integrated into passwords screen

amddg44 committed:
SyncPromoView icon

amddg44 committed:
Bookmarks search bar refactored from storyboard to code & embedded in uiview

amddg44 committed:
Add & removed SyncPromoView to headerView below searchBar dynamically

amddg44 committed:
Search bar placeholder text

amddg44 committed:
Search bar placeholder text

amddg44 committed:
New row added to Sync screen to get DDG on other devices / platforms

amddg44 committed:
New get ddg on other devices screen

amddg44 committed:
String updates

amddg44 committed:
Debug reset option added

amddg44 committed:
Add new get DDG button to DeviceConnectedView

amddg44 committed:
SyncSettingsViewModel updated for new button actions

amddg44 committed:
Linter violation fix

amddg44 committed:
Update sync asset names that xcode was complaining conflict with primary app assets

amddg44 committed:
Share action implemented

amddg44 committed:
Unit tests updated + added to

amddg44 committed:
Fixed typo

amddg44 committed:
Removed unnecessary SyncPromoViewModelType

amddg44 committed:
Improvement to share action to only include message for mail & message actions

amddg44 committed:
Fix for layout issue when auth not required on screen load

amddg44 committed:
Merge branch 'main' into anya/sync-touchpoints-promotion

graeme reviewed (approved):
Looks good! Just one small thing: On smaller devices, the word wrapping looks a bit weird. Just needs centre alignment. Otherwise all good :)

![Simulator Screenshot - iPhone SE (3rd generation) - 2024-08-23 at 12 40 38](https://github.com/user-attachments/assets/33be65f8-f3c7-4b2b-8976-fd11004aa238)

amddg44 committed:
Pixels for new Sync promos (#3269)

Task/Issue URL: https://app.asana.com/0/1142021229838617/1208017419957479/f
Tech Design URL:
CC:

Description:
Pixels for the new Sync Promo feature + new platform links in Sync screen

amddg44 committed:
Make text center aligned

amddg44 committed:
Copy updates from ship review

amddg44 committed:
Remove temporary pixel

amddg44 committed:
Merge remote-tracking branch 'origin/main' into anya/sync-touchpoints-promotion

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
#	DuckDuckGo/BookmarksViewController.swift
#	DuckDuckGo/MainViewController+Segues.swift

amddg44 committed:
Merged Package.resolved

amddg44 committed:
Bump BSK ref

amddg44 committed:
Merge remote-tracking branch 'origin/main' into anya/sync-touchpoints-promotion

# Conflicts:
#	Core/FeatureFlag.swift
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

amddg44 committed:
Bump BSK ref

amddg44 committed:
Set BSK release
